server/src/Routes/Checkout.hs

```diff
@@ -840,16 +804,48 @@ anonymousPlaceOrderRoute remoteHost forwardedFor realIP = validate >=> \paramete
         reduceQuantities orderId
         orderLines <- selectList [OrderLineItemOrderId ==. orderId] []
         products <- getCheckoutProducts orderId
+        guestToken <- generateGuestToken orderId
         return
             ( orderId
             , AnonymousPlaceOrderData
                 { apodOrderId = orderId
+                , apodGuestToken = guestToken
                 , apodLines = orderLines
                 , apodProducts = products
                 }
             )
     runDB $ enqueuePostOrderTasks orderId
     pure orderData
+    where
+        createAnonymousCustomer parameters = do
+            let email = apopEmail parameters
+                shippingParam = apopShippingAddress parameters
+                billingParam = apopBillingAddress parameters
+                makeAvalaraCustomer = createAvalaraCustomer email
+                    $ fromMaybe shippingParam billingParam
+
+            (Entity cId c) <- lift $ createNewCustomer True RegistrationParameters
+                    { rpEmail = email
+                    , rpPassword = "00000000"
```

Comment:
I wonder if using the hardcoded password may cause issues?
AFAIR, we do not check if a user is ephemeral during the `login` and one may _potentially_ guess the password by the known hardcoded hash

Reply:
> AFAIR, we do not check if a user is ephemeral during the login

We do:
```
validatePassword email password =
  ..............
            if | customerEphemeral customer
                -> hashAnyways $ throwIO NoCustomer
```

But your suggestion about nullable password should be more natural there.

Reply:
Oh, I see, I missed changes in `validatePassword`, thanks for the clarification